languagetool-language-modules/en/src/main/java/org/languagetool/rules/en/EnglishWordRepeatBeginningRule.java

```diff
@@ -44,35 +45,84 @@ public String getId() {
     return "ENGLISH_WORD_REPEAT_BEGINNING_RULE";
   }
   
-  private static final Set<String> ADVERBS = new HashSet<>();
-  static {
-    ADVERBS.add("Additionally");
-    ADVERBS.add("Besides");
-    ADVERBS.add("Furthermore");
-    ADVERBS.add("Moreover");
-  }
+	// adverbs used to add to what the previous sentence mentioned
+	private static final Set<String> ADD_ADVERBS = new HashSet<>();
 
-  @Override
-  public boolean isException(String token) {
-    return super.isException(token) || token.equals("The") || token.equals("A") || token.equals("An");
-  }
+	// adverbs used to express contrast to what the previous sentence mentioned
+	private static final Set<String> CONTRAST_ADVERBS = new HashSet<>();
 
-  @Override
-  protected boolean isAdverb(AnalyzedTokenReadings token) {
-    return ADVERBS.contains(token.getToken());
-  }
+	// adverbs used to express emphasis to what the previous sentence mentioned
+	private static final Set<String> EMPHASIS_ADVERBS = new HashSet<>();
 
-  @Override
-  protected List<String> getSuggestions(AnalyzedTokenReadings token) {
-    if (token.hasPosTag("PRP")) {
-      String tok = token.getToken();
-      String adaptedToken = tok.equals("I") ? tok : tok.toLowerCase();
-      return Arrays.asList(
-              "Furthermore, " + adaptedToken,
-              "Likewise, " + adaptedToken,
-              "Not only that, but " + adaptedToken);
-    }
-    return Collections.emptyList();
-  }
+	// adverbs used to explain what the previous sentence mentioned
+	private static final Set<String> EXPLAIN_ADVERBS = new HashSet<>();
+
+	static {
+		// based on https://www.pinterest.com/pin/229542912245527548/
+		ADD_ADVERBS.add("Additionally");
+		ADD_ADVERBS.add("Besides");
+		ADD_ADVERBS.add("Furthermore");
+		ADD_ADVERBS.add("Moreover");
+		ADD_ADVERBS.add("Also");
+		CONTRAST_ADVERBS.add("Unlike");
+		CONTRAST_ADVERBS.add("Nevertheless");
+		CONTRAST_ADVERBS.add("Nonetheless");
+		CONTRAST_ADVERBS.add("Despite");
```

Comment:
Thank you, @giorgossideris! This is a cool rule!

Just one thing: Could you take out "Unlike" and "Despite"? They are not adverbs and, most importantly, they cannot be used as sentence adverbs at the beginning of a sentence ("Nevertheless, …" is okay, "Despite, …" and "Unlike, …" are not). So if someone writes several sentences starting with "Nonetheless", they should not be offered as possible replacements.

They could go into a different set, e.g. CONTRAST_PREPOSITIONS…

Reply:
@udomai That's a pretty good observation, I didn't think of that. I will make the appropriate changes!